src/main/java/com/google/api/generator/gapic/composer/ServiceClientClassComposer.java

```diff
@@ -154,7 +157,9 @@ public GapicClass generate(Service service, Map<String, Message> messageTypes) {
     methods.addAll(createStaticCreatorMethods(service, types));
     methods.addAll(createConstructorMethods(service, types, hasLroClient));
     methods.addAll(createGetterMethods(service, types, hasLroClient));
-    methods.addAll(createServiceMethods(service, messageTypes, types));
+    methods.addAll(
+        createServiceMethods(
+            service, messageTypes, types, types.get(getClientClassName(service.name()))));
```

Comment:
Instead of passing client type all the way from here, let's pass the service name from inside createServiceMethods, then get the clientType in createMethodVariants.